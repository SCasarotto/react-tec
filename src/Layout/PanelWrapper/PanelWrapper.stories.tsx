import React from 'react';

import { Story, Meta } from '@storybook/react';

import { PanelWrapper, PanelWrapperProps } from './PanelWrapper';

export default {
  title: 'Layout/PanelWrapper',
  component: PanelWrapper,
  argTypes: {},
} as Meta;

const Template: Story<PanelWrapperProps> = (args) => <PanelWrapper {...args} />;

export const Default = Template.bind({});
Default.args = {};
