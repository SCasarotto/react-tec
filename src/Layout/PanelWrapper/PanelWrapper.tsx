import React from 'react';

import { Container } from './styledComponents';

export interface PanelWrapperProps
  extends React.HTMLAttributes<HTMLDivElement> {}
/**
 *
 * PanelWrapper simple layout component meant to be used as the parent of `Panel`. Below is the code that wraps this page.
 *
 * ```
 * <PanelWrapper>
 *		{ ... }
 * </PanelWrapper>
 * ```
 *
 */
export const PanelWrapper: React.FC<PanelWrapperProps> = (props) => {
  const { className = '', ...rest } = props;
  return <Container className={`tec-panelWrapper ${className}`} {...rest} />;
};
