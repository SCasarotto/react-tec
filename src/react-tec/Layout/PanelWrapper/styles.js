export default {
	container : {
		display 	    	: 'flex',
		flexFlow 	    	: 'row wrap',
		justifyContent		: 'space-between',
		alignItems      	: 'flex-start',

		paddingLeft 		: 30,
		paddingRight 		: 30,
		paddingTop 			: 30,
		paddingBottom 		: 30,

		'@media (max-width: 800px)' : {
			paddingLeft 		: 25,
			paddingRight 		: 25,
			paddingTop 			: 25,
			paddingBottom 		: 25,
		},
		'@media (max-width: 650px)' : {
			paddingLeft 		: 20,
			paddingRight 		: 20,
			paddingTop 			: 20,
			paddingBottom 		: 20,
		},
		'@media (max-width: 450px)' : {
			paddingLeft 		: 15,
			paddingRight 		: 15,
			paddingTop 			: 15,
			paddingBottom 		: 15,
		}
	},
}
