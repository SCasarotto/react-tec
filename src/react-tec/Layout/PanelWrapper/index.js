import PanelWrapper from './PanelWrapper';

export default PanelWrapper;
