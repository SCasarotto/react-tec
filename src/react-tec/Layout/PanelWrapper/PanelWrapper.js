//
//TE Version 0.2.0
//

import React, { Component } from 'react'
import Radium from 'radium'

import styles from './styles'

class PannelWrapper extends Component {
	render() {
		const { style } = this.props
		return <div style={{ ...styles.container, ...style }}>{this.props.children}</div>
	}
}

PannelWrapper.propTypes = {}

PannelWrapper.defaultProps = {}

export default Radium(PannelWrapper)
