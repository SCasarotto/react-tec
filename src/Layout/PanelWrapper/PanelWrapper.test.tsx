import React from 'react';

import { render } from '@testing-library/react';

import { PanelWrapper } from './PanelWrapper';

describe('PanelWrapper', () => {
  // Renders Without Error
  it('matches snapshot', () => {
    const { asFragment } = render(<PanelWrapper />);
    expect(asFragment()).toMatchSnapshot();
  });

  // Class Name
  it('accepts custom className', () => {
    const { container } = render(<PanelWrapper className='myClassName' />);
    expect(container.firstChild).toHaveClass('myClassName');
  });
});
