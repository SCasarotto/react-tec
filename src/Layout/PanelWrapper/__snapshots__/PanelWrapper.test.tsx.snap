// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelWrapper matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 30px;
}

@media (max-width:800px) {
  .c0 {
    padding: 25px;
  }
}

@media (max-width:650px) {
  .c0 {
    padding: 20px;
  }
}

@media (max-width:450px) {
  .c0 {
    padding: 15px;
  }
}

<div
    class="c0 tec-panelWrapper "
  />
</DocumentFragment>
`;
